### PR TITLE
Make ConvContext be data type-aware.

### PR DESCRIPTION
**Pull request by whchung**
And pick such information in vectorization logic.

### CODE REVIEW

File: mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h [hunk 1]
```diff
@@ -20,6 +20,8 @@
 
 namespace mlir {
 
+enum class DataType { f32, f16, bf16 };
```
Comment (jerryyin) [inline mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h]: I feel like it make better sense if you use the existing type infrastructure to handle this: see `mlir/lib/IR/Types.cpp`

That should give you a free implementation to get bitwidth and with which you can decide vectorization size based on bitwidth.

Comment (whchung) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h]: @jerryyin is this `ConvContext` used by external clients? if not, I'll switch to MLIR `Type`.

Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h]: It's hidden behind and is not exposed in the API. It's up to us how this is implemented.

Comment (whchung) [reply mlir/include/mlir/Dialect/MIOpen/Tuning/ConvContext.h]: Got it. Let me revise the implementation, using `mlir::Type`.

**Review by jerryyin — APPROVED** [review]

**Comment by whchung** [comment]
jenkins: retest this please.

**Comment by whchung** [comment]
@okakarpa wondering why `mli-ci.amd.com` target for this PR can't work? `rocm-head-ci` just completed without issue.

**Comment by okakarpa** [comment]
> @okakarpa wondering why `mli-ci.amd.com` target for this PR can't work? `rocm-head-ci` just completed without issue.

seems to be issue with the node i have retriggered the job

http://ml-ci.amd.com:21096/job/MLIR/job/mlir/view/change-requests/job/PR-276/4/console

**Pull request merged**